### PR TITLE
[Datasets] Use SPREAD scheduling strategy and turn off Parquet sampling by default

### LINKED ISSUE #26966: [Core][Nightly] datasets_ingest_train_infer failing
### What happened + What you expected to happen

failing test: https://console.anyscale.com/o/anyscale-internal/projects/prj_2xR6uT6t7jJuu1aCwWMsle/clusters/ses_rFHQEVVAA3KZgi4apbnTAkcd?command-history-section=command_history
```
Step 1: Dropping nulls, creating new_col, updating feature_1

Map_Batches:  99%|█████████████████████████████████████████████████████████████████████████████████████▏| 396/400 [00:19<00:00, 78.36it/s](pid=gcs_server) E0725 08:34:14.326870515     253 chttp2_transport.cc:2846]   keepalive_ping_end state error: 0 (expect: 1)
(raylet, ip=10.0.0.128) [2022-07-25 08:35:14,805 C 157 212] (raylet) gcs_rpc_client.h:537:  Check failed: absl::ToInt64Seconds(absl::Now() - gcs_last_alive_time_) < ::RayConfig::instance().gcs_rpc_server_reconnect_timeout_s() Failed to connect to GCS within 60 seconds
(raylet, ip=10.0.0.128) *** StackTrace Information ***
(raylet, ip=10.0.0.128) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47af5a) [0x55e3772d0f5a] ray::operator<<()
(raylet, ip=10.0.0.128) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47ca52) [0x55e3772d2a52] ray::SpdLogMessage::Flush()
(raylet, ip=10.0.0.128) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47cd67) [0x55e3772d2d67] ray::RayLog::~RayLog()
(raylet, ip=10.0.0.128) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x353253) [0x55e3771a9253] ray::rpc::GcsRpcClient::CheckChannelStatus()
(raylet, ip=10.0.0.128) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x3536c3) [0x55e3771a96c3] boost::asio::detail::wait_handler<>::do_complete()
(raylet, ip=10.0.0.128) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x9896eb) [0x55e3777df6eb] boost::asio::detail::scheduler::do_run_one()
(raylet, ip=10.0.0.128) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x98aeb1) [0x55e3777e0eb1] boost::asio::detail::scheduler::run()
(raylet, ip=10.0.0.128) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x98b0e0) [0x55e3777e10e0] boost::asio::io_context::run()
(raylet, ip=10.0.0.128) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x9d9fa0) [0x55e37782ffa0] execute_native_thread_routine
(raylet, ip=10.0.0.128) /lib/x86_64-linux-gnu/libpthread.so.0(+0x76db) [0x7f108a3366db] start_thread
(raylet, ip=10.0.0.128) /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f108951861f] __clone
(raylet, ip=10.0.0.128) 
Raylet is terminated: ip=10.0.0.128, id=ab515a6f7fd0794fab61dc4ac46742175cdff63997306a6de835bfb6. Termination is unexpected. Possible reasons include: (1) SIGKILL by the user or system OOM killer, (2) Invalid memory access from Raylet causing SIGSEGV or SIGBUS, (3) Other termination signals. Last 20 lines of the Raylet logs:
    [state-dump]        AgentManagerService.grpc_server.RegisterAgent - 1 total (0 active), CPU time: mean = 291.934 us, total = 291.934 us
    [state-dump]        NodeInfoGcsService.grpc_client.RegisterNode - 1 total (0 active), CPU time: mean = 489.663 us, total = 489.663 us
    [state-dump]        NodeInfoGcsService.grpc_client.GetInternalConfig - 1 total (0 active), CPU time: mean = 1.014 s, total = 1.014 s
    [state-dump] DebugString() time ms: 1
    [state-dump] 
    [state-dump] 
    [2022-07-25 08:35:14,805 C 157 212] (raylet) gcs_rpc_client.h:537:  Check failed: absl::ToInt64Seconds(absl::Now() - gcs_last_alive_time_) < ::RayConfig::instance().gcs_rpc_server_reconnect_timeout_s() Failed to connect to GCS within 60 seconds
    *** StackTrace Information ***
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47af5a) [0x55e3772d0f5a] ray::operator<<()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47ca52) [0x55e3772d2a52] ray::SpdLogMessage::Flush()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47cd67) [0x55e3772d2d67] ray::RayLog::~RayLog()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x353253) [0x55e3771a9253] ray::rpc::GcsRpcClient::CheckChannelStatus()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x3536c3) [0x55e3771a96c3] boost::asio::detail::wait_handler<>::do_complete()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x9896eb) [0x55e3777df6eb] boost::asio::detail::scheduler::do_run_one()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x98aeb1) [0x55e3777e0eb1] boost::asio::detail::scheduler::run()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x98b0e0) [0x55e3777e10e0] boost::asio::io_context::run()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x9d9fa0) [0x55e37782ffa0] execute_native_thread_routine
    /lib/x86_64-linux-gnu/libpthread.so.0(+0x76db) [0x7f108a3366db] start_thread
    /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f108951861f] __clone
    

(pid=gcs_server) [2022-07-25 08:35:34,426 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
Raylet is terminated: ip=10.0.0.150, id=717b4ae7fe19f51c9d16947065f5a1144e034fa386e3da99a3fbda50. Termination is unexpected. Possible reasons include: (1) SIGKILL by the user or system OOM killer, (2) Invalid memory access from Raylet causing SIGSEGV or SIGBUS, (3) Other termination signals. Last 20 lines of the Raylet logs:
    [state-dump]        NodeInfoGcsService.grpc_client.RegisterNode - 1 total (0 active), CPU time: mean = 431.974 us, total = 431.974 us
    [state-dump]        JobInfoGcsService.grpc_client.AddJob - 1 total (0 active), CPU time: mean = 41.452 us, total = 41.452 us
    [state-dump] DebugString() time ms: 1
    [state-dump] 
    [state-dump] 
    [2022-07-25 08:35:36,203 W 273 337] (raylet) node_manager.cc:158: Last heartbeat was sent 8987 ms ago. There might be resource pressure on this node. If heartbeat keeps lagging, this node can be marked as dead mistakenly.
    [2022-07-25 08:35:40,492 C 273 339] (raylet) gcs_rpc_client.h:537:  Check failed: absl::ToInt64Seconds(absl::Now() - gcs_last_alive_time_) < ::RayConfig::instance().gcs_rpc_server_reconnect_timeout_s() Failed to connect to GCS within 60 seconds
    *** StackTrace Information ***
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47af5a) [0x55da47c05f5a] ray::operator<<()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47ca52) [0x55da47c07a52] ray::SpdLogMessage::Flush()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47cd67) [0x55da47c07d67] ray::RayLog::~RayLog()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x353253) [0x55da47ade253] ray::rpc::GcsRpcClient::CheckChannelStatus()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x3536c3) [0x55da47ade6c3] boost::asio::detail::wait_handler<>::do_complete()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x9896eb) [0x55da481146eb] boost::asio::detail::scheduler::do_run_one()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x98aeb1) [0x55da48115eb1] boost::asio::detail::scheduler::run()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x98b0e0) [0x55da481160e0] boost::asio::io_context::run()
    /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x9d9fa0) [0x55da48164fa0] execute_native_thread_routine
    /lib/x86_64-linux-gnu/libpthread.so.0(+0x76db) [0x7fbe6cbd56db] start_thread
    /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7fbe6bdb761f] __clone
    

(pid=gcs_server) [2022-07-25 08:35:41,696 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:41,709 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(raylet) [2022-07-25 08:35:40,492 C 273 339] (raylet) gcs_rpc_client.h:537:  Check failed: absl::ToInt64Seconds(absl::Now() - gcs_last_alive_time_) < ::RayConfig::instance().gcs_rpc_server_reconnect_timeout_s() Failed to connect to GCS within 60 seconds
(raylet) *** StackTrace Information ***
(raylet) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47af5a) [0x55da47c05f5a] ray::operator<<()
(raylet) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47ca52) [0x55da47c07a52] ray::SpdLogMessage::Flush()
(raylet) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x47cd67) [0x55da47c07d67] ray::RayLog::~RayLog()
(raylet) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x353253) [0x55da47ade253] ray::rpc::GcsRpcClient::CheckChannelStatus()
(raylet) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x3536c3) [0x55da47ade6c3] boost::asio::detail::wait_handler<>::do_complete()
(raylet) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x9896eb) [0x55da481146eb] boost::asio::detail::scheduler::do_run_one()
(raylet) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x98aeb1) [0x55da48115eb1] boost::asio::detail::scheduler::run()
(raylet) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x98b0e0) [0x55da481160e0] boost::asio::io_context::run()
(raylet) /home/ray/anaconda3/lib/python3.7/site-packages/ray/core/src/ray/raylet/raylet(+0x9d9fa0) [0x55da48164fa0] execute_native_thread_routine
(raylet) /lib/x86_64-linux-gnu/libpthread.so.0(+0x76db) [0x7fbe6cbd56db] start_thread
(raylet) /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7fbe6bdb761f] __clone
(raylet) 
(pid=gcs_server) [2022-07-25 08:35:42,584 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:42,584 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:43,587 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:43,588 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:44,590 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:44,590 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:45,593 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:45,593 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:46,595 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:46,595 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:47,598 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:47,599 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:48,601 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
(pid=gcs_server) [2022-07-25 08:35:48,601 E 217 217] (gcs_server) gcs_server.cc:295: Failed to get the resource load: GrpcUnavailable: RPC Error message: failed to connect to all addresses; RPC Error details: 
Traceback (most recent call last):
  File "ray_sgd_training.py", line 524, in <module>
    read_dataset(data_path)
  File "ray_sgd_training.py", line 47, in preprocess_train_data
    return self._preprocess(ds, False)
  File "ray_sgd_training.py", line 75, in _preprocess
    ds = ds.map_batches(batch_transformer, batch_format="pandas")
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/dataset.py", line 499, in map_batches
    return Dataset(plan, self._epoch, self._lazy)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/dataset.py", line 207, in __init__
    self._plan.execute(allow_clear_input_blocks=False)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/_internal/plan.py", line 309, in execute
    blocks, clear_input_blocks, self._run_by_consumer
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/_internal/plan.py", line 672, in __call__
    fn_constructor_kwargs=self.fn_constructor_kwargs,
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/_internal/compute.py", line 113, in _apply
    results = map_bar.fetch_until_complete(refs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/_internal/progress_bar.py", line 75, in fetch_until_complete
    for ref, result in zip(done, ray.get(done)):
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/worker.py", line 2233, in get
    values, debugger_breakpoint = worker.get_objects(object_refs, timeout=timeout)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/worker.py", line 668, in get_objects
    object_refs, self.current_task_id, timeout_ms
  File "python/ray/_raylet.pyx", line 1219, in ray._raylet.CoreWorker.get_objects
  File "python/ray/_raylet.pyx", line 179, in ray._raylet.check_status
ray.exceptions.RaySystemError: System error: Connection reset by peer
```

### Versions / Dependencies

latest ray

### Reproduction script

run nightly test

### Issue Severity

High: It blocks me from completing my task.

### LINKED ISSUE #26995: [AIR] Significant data reading regression in Ray cluster from xgboost 100GB test
## Issues observed 

1) We're not able to evenly distribute read tasks evenly across a cluster anymore with significant skew that lead to imbalanced memory pattern
2) Significantly increased memory cost on headnode that easily lead to OOM

Regression happend between ray commit `8ecd928c34db0b23e4aa2a4ea0c8cff25c37b413` and `aadd82dcbd6bb0a8083550ef3edf39c98bf08ce0`, roughly in past 3 days from now.

## Good nightly release run
Link to good nightly release run: https://console.anyscale-staging.com/o/anyscale-internal/projects/prj_mWECugke9RzMh79BZQqeykjN/clusters/ses_r2ZVKt7AzSsY1seXN5hqene4?command-history-section=command_history

Logs: https://gist.github.com/jiaodong/6fd5728e35b23e6f78c4c6049b754d09

Pip freeze: https://gist.github.com/jiaodong/7d84cc4a73eb80d2a7b40dc31b507138

## Bad nightly release run 
Link to bad nightly release run: https://console.anyscale-staging.com/o/anyscale-internal/projects/prj_mWECugke9RzMh79BZQqeykjN/clusters/ses_egWTLrS2PYDUJsQYzeKjiqzP?command-history-section=head_start_up_log

Logs: https://gist.github.com/jiaodong/73d716ea47a9319aa11e629719a1d735

Pip freeze: https://gist.github.com/jiaodong/8ea91f79b7af6602479a970b751a8679

## Good run memory usage metrics 
Screen Shot 2022-07-25 at 5 15 02 PM

## Bad run memory usage metrics 
Screen Shot 2022-07-25 at 5 15 43 PM

Corresponding Ray dashboard stats

## Good run ray dashboard
Screen Shot 2022-07-25 at 5 14 39 PM

## Bad run ray dashboard
Screen Shot 2022-07-25 at 5 15 38 PM

### Versions / Dependencies

On master

### Reproduction script

Re-run `xgboost_benchmark.py`

https://sourcegraph.com/github.com/ray-project/ray/-/blob/release/air_tests/air_benchmarks/workloads/xgboost_benchmark.py

with 100GB data on 10 node ray cluster 

```yaml
max_workers: 9
--
  |  
  | head_node_type:
  | name: head_node
  | instance_type: m5.4xlarge
  |  
  | worker_node_types:
  | - name: worker_node
  | instance_type: m5.4xlarge
  | max_workers: 9
  | min_workers: 9
  | use_spot: false
  |  
  | aws:
  | BlockDeviceMappings:
  | - DeviceName: /dev/sda1
  | Ebs:
  | Iops: 5000
  | Throughput: 1000
  | VolumeSize: 1000
  | VolumeType: gp3

```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by c21**
## Why are these changes needed?

The Parquet file sampling PR (https://github.com/ray-project/ray/pull/26868) caused nightly test regression - https://github.com/ray-project/ray/issues/26995 . Turning off the feature by default, and keep debugging the root cause, to unblock nightly test.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]

**Review by clarkzinzow — APPROVED** [review]

**Comment by c21** [comment]
Hi @ericl and @clarkzinzow - make some more change to the PR, and could you help take a look again during daytime? Thanks. More context in https://github.com/ray-project/ray/issues/26995#issuecomment-1196283417.

**Comment by scv119** [comment]
huh seems the sampling is also causing https://github.com/ray-project/ray/issues/26966 to regress. 

It's probably because we creates way more objects that the owner can handle and eventually we crashed.

**Review by scv119 — APPROVED** [review]

**Comment by scv119** [comment]
Chatted with @c21 offline, if we turn this off the SPREAD and PARQUET_ENCODING_RATIO_ESTIMATE_MIN_NUM_SAMPLES change won't be effective.

File: python/ray/data/context.py [hunk 1]
```diff
@@ -66,7 +66,7 @@
 DEFAULT_USE_POLARS = False
 
 # Whether to estimate in-memory decoding data size for data source.
-DEFAULT_DECODING_SIZE_ESTIMATION_ENABLED = True
+DEFAULT_DECODING_SIZE_ESTIMATION_ENABLED = False
```
Comment (c21) [inline python/ray/data/context.py:69]: Thanks @scv119 for flagging the issue. FYI this is the code disabling sampling logic completely. So below ratio + strategy change is non-effect given this change.

File: python/ray/data/datasource/parquet_datasource.py [hunk 2]
```diff
@@ -314,7 +314,13 @@ def _estimate_files_encoding_ratio(self) -> float:
         futures = []
         for idx, sample in enumerate(file_samples):
             # Sample i-th row group in i-th file.
-            futures.append(sample_piece.remote(_SerializedPiece(sample), idx))
+            # Use SPREAD scheduling strategy to avoid packing many sampling tasks on
+            # same machine to cause OOM issue, as sampling can be memory-intensive.
```
Comment (ericl) [inline python/ray/data/datasource/parquet_datasource.py:318]: Why does this cause an OOM? We should read a rowbatch at most which should be pretty small right? Are we using the right read APIs?

Comment (c21) [reply python/ray/data/datasource/parquet_datasource.py:318]: @ericl - yes that's the expectation. I actually found Arrow reader allocated much more memory than needed in https://github.com/ray-project/ray/issues/26995#issuecomment-1196283417 . The current API `ParquetFileFragment.head(num_rows, batch_size=num_rows)` indeed gives us pretty accurate in-memory data size. So I am not aware of other better APIs, but I will create a Arrow JIRA tomorrow to ask Arrow experts.

**Pull request merged**